opendatahub-io/odh-dashboard
# merge 'f/byon' to incubation

## Description


Add custom notebook phase 1 work to incubation branch

## How Has This Been Tested?



Image locations you could use to test:
- quay.io/opendatahub/workbench-images:jupyter-minimal-ubi8-python-3.8-pr-89 (valid)
- docker.io/invalid/invalid (invalid)
- quay.io/opendatahub/workbench-images:invalid (invalid)

Test instructions:
1. Go to Settings -> Notebook image settings
2. Click on `Import notebook image`, and try to create an image using the locations above
3. Try to add packages and software to the image, you can add anything you want
4. Test the keyboard interactions when inputting packages/software, when you edit a row, hit `Enter` could help you to save the row and move to the next row. If you are current on the last row, it will create a new row. Hit `Esc` could exit the edit mode and abandon the current change on the row.

Test checklist:
- [ ] Make sure you could import a new image
- [ ] Make sure the sort and search of the table work
- [ ] The invalid images have an error state with the `Enable` toggle locked and disabled
- [ ] When you input the image location with an invalid format, make sure the modal won't close but show an error alert inside
- [ ] Make sure you can successfully update and delete the image

## Test Impact



## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
